src/main/java/com/sublinks/sublinksapi/person/services/PersonService.java

```diff
@@ -189,6 +191,9 @@ public void createPerson(final Person person) {
 
     person.setPassword(passwordEncoder.encode(person.getPassword()));
 
+    final String userActorId = baseUrlUtil.getBaseUrl() + "/u/" + person.getName();
```

Comment:
This might be ok for now. However, I was thinking about our discussion earlier today and we might be able to do something with the user creation flow between API and federation. Eg:

- User registers for account on frontend/API
- API creates account _without actor_id_
- API sends message to federation to register user (has to happen anyway)
- Federation returns a message either the actor_id (this part is new)
- API saves actor_id to DB

There might need to be some kind of them actor_id in between though I guess in order to ensure the frontend can function?

Reply:
What i see is it will crash the default lemmy ui as seen. We can make it temporarily return a assumed actor id.

Reply:
Like a mock actor_id for private instances that won't be using federation?

Reply:
We could also set like a global env config USE_FEDERATION that could make the api optionally set the actor_ids and the Queue stuff 🤔

Reply:
I would just say, the api assumes the format DOMAIN/user/USERNAME

And if the federation says NO!!! then api changes it to it.

ap_id and actor_id is required on lemmy to be a url. And cant be empty for a seconds / minutes until the federation processed that user/community/comment/post